plugins/generator-1.18.2/forge-1.18.2/aitasks/attack_with_range.java.ftl

```diff
@@ -0,0 +1,10 @@
+<#include "aiconditions.java.ftl">
+this.goalSelector.addGoal(${customBlockIndex+1}, new MeleeAttackGoal(this, ${field$speed}, ${field$longmemory?lower_case}) {
+
+	@Override protected double getAttackReachSqr(LivingEntity entity) {
+		return ${field$range} * 4;

```

Comment:
What is the purpose of this multiplication and why does this one have it set to 4 and 1.19.2 to 3?

Reply:
@KlemenDEV also updated name to include "in blocks"

Reply:
> The purpose is that it doesn't use block range

What do you mean by block range?

Reply:
I mean that the original measurement that the procedure comes in with is not "in blocks", it's in a different measurement (no idea exactly which one it is), and multiplying it by 4 makes it match the distance in blocks, so 1 x 4 = 1 block @KlemenDEV

Reply:
No need to tag me in each comment. If you look into the vanilla code, this is not true. Can you show me the source of this?

I have checked whole ranged attack pipeline and standard units are used.

I think you are confusing meaning of word "squared" for squared distance, as this simplifies euclidian distance calculation as square root is not needed in this case

Reply:
Hmm when manually testing the range, it did not seem like that at all; Will take your word for it

Reply:
You probably need to return range to the power of two, not directly. That is why you probably thought it has strange units.

> I think you are confusing meaning of word "squared" for squared distance, as this simplifies euclidian distance calculation as square root is not needed in this case

Reply:
It is called getAttackReachSqr, not getAttackReach

Reply:
Alright, so do you want me to change the value to range * range?

Reply:
Yes, this should work, but ofc try it out also

Reply:
Alright, will test the changes out in a few hours when I get some time, will then commit here (once the commit is here, it means I tested it)

Reply:
You can multiply inside FTL print for cleaner code: ${field$range * field$range}

Reply:
That's cool to know, ty!

Reply:
Alr, no, doing ${field$range * field$range} does not work at all

Reply:
Maybe you would need to convert them to numbers first, field value is probably string

`${field$range?number * field$range?number}`

Reply:
Will test that out in a few hours

Reply:
That does work!